packages/framework/ai-collab/src/explicit-strategy/index.ts

```diff
@@ -0,0 +1,362 @@
+}
+
+/**
+ * Prompts the provided LLM client to generate valid tree edits.
+ * Applies those edits to the provided tree branch before returning.
+ *
+ * @remarks
+ * - Optional root nodes are not supported
+ * - Primitive root nodes are not supported
+ *
+ * @internal
+ */
+export async function generateTreeEdits<TSchema extends ImplicitFieldSchema>(
+	options: GenerateTreeEditsOptions<TSchema>,
+): Promise<GenerateTreeEditsSuccessResponse | GenerateTreeEditsErrorResponse> {
+	const idGenerator = new IdGenerator();
+	const editLog: EditLog = [];
+	let editCount = 0;
+	let sequentialErrorCount = 0;
+
+	const isRootNode = Tree.parent(options.treeNode) === undefined;
+	const simpleSchema = isRootNode
+		? getSimpleSchema(normalizeFieldSchema(options.treeView.schema).allowedTypes)
+		: getSimpleSchema(Tree.schema(options.treeNode));
+
+	const tokenUsage = { inputTokens: 0, outputTokens: 0 };
+
+	try {
+		for await (const edit of generateEdits(
+			options,
+			simpleSchema,
+			idGenerator,
+			editLog,
+			options.limiters?.tokenLimits,
+			tokenUsage,
+		)) {
+			try {
+				const result = applyAgentEdit(
+					options.treeView,
+					options.treeNode,
+					edit,
+					idGenerator,
+					simpleSchema.definitions,
+					options.validator,
+				);
+				const explanation = result.explanation; // TODO: describeEdit(result, idGenerator);
+				editLog.push({ edit: { ...result, explanation } });
+				sequentialErrorCount = 0;
+			} catch (error: unknown) {
+				if (error instanceof Error) {
+					sequentialErrorCount += 1;
+					editLog.push({ edit, error: error.message });
+					DEBUG_LOG?.push(`Error: ${error.message}`);
+				} else {
+					throw error;
+				}
+			}
+
+			if (options.limiters?.abortController?.signal.aborted === true) {
+				return {
+					status: "failure",
+					errorMessage: "aborted",
+					tokenUsage,
+				};
+			}
+
+			if (
+				sequentialErrorCount >
+				(options.limiters?.maxSequentialErrors ?? Number.POSITIVE_INFINITY)
+			) {
+				return {
+					status: "failure",
+					errorMessage: "tooManyErrors",
+					tokenUsage,
+				};
+			}
+
+			if (++editCount >= (options.limiters?.maxModelCalls ?? Number.POSITIVE_INFINITY)) {
+				return {
+					status: "failure",
+					errorMessage: "tooManyModelCalls",
+					tokenUsage,
+				};
+			}
+		}
+	} catch (error: unknown) {
+		if (error instanceof Error) {
+			DEBUG_LOG?.push(`Error: ${error.message}`);
+		}
+		if (error instanceof TokenLimitExceededError) {
+			return {
+				status: "failure",
+				errorMessage: "tokenLimitExceeded",
+				tokenUsage,
+			};
+		}
+		throw error;
+	}
+
+	if (options.dumpDebugLog ?? false) {
+		console.log(DEBUG_LOG.join("\n\n"));
+		DEBUG_LOG.length = 0;
+	}
+
+	return {
+		status: "success",
+		tokenUsage,
+	};
+}
+
+interface ReviewResult {
+	goalAccomplished: "yes" | "no";
+}
+
+/**
+ * Generates a single {@link TreeEdit} from an LLM.
+ * The design of this async generator function is such that which each iteration of this functions values,
```

Comment:
```suggestion
 * @remarks
 * The design of this async generator function is such that with each iteration of this function's values,
```

Reply:
Updated in commit 6eb662389384cc2360c519307d3b33d3a7a02309